packages/context/test/unit/context.ts

```diff
@@ -63,10 +63,39 @@ describe('Context', () => {
     it('throws a helpful error when the binding is async', () => {
       ctx.bind('foo').toDynamicValue(() => Promise.resolve('bar'));
       expect(() => ctx.getSync('foo')).to.throw(/foo.*async/);
-   });
+    });
+  });
+
+  describe('context.bind().toProvider()', () => {
+    it('binds a provider class to a context', () => {
+      ctx.bind('foo').toProvider(MyProvider);
```

Comment:
I don't think this isn't the usage we are looking for. My understanding is that we want to use the key specified by the provider, therefore this line should be closer to this:

```ts
ctx.bindProvider(MyProvider);
```

I am imagining `bindProvider` can be implemented as follows:

```ts
bindProvider(provider) {
  this.bind(provider.key).toProvider(provider);
}
```